drivers/spi/spi_context.h

```diff
@@ -154,33 +154,32 @@ static inline void spi_context_cs_control(struct spi_context *ctx, bool on)
 	}
 }
 
-static inline void spi_context_buffers_setup(struct spi_context *ctx,
-					     const struct spi_buf *tx_bufs,
-					     size_t tx_count,
-					     struct spi_buf *rx_bufs,
-					     size_t rx_count,
-					     u8_t dfs)
+static inline
+void spi_context_buffers_setup(struct spi_context *ctx,
+			       const struct spi_buf_array *tx_bufs,
+			       const struct spi_buf_array *rx_bufs,
+			       u8_t dfs)
 {
-	SYS_LOG_DBG("tx_bufs %p (%zu) - rx_bufs %p (%zu) - %u",
-		    tx_bufs, tx_count, rx_bufs, rx_count, dfs);
-
-	ctx->current_tx = tx_bufs;
-	ctx->tx_count = tx_count;
-	ctx->current_rx = rx_bufs;
-	ctx->rx_count = rx_count;
+	SYS_LOG_DBG("tx_bufs %p - rx_bufs %p - %u", tx_bufs, rx_bufs, dfs);
 
 	if (tx_bufs) {
-		ctx->tx_buf = tx_bufs->buf;
-		ctx->tx_len = tx_bufs->len / dfs;
+		ctx->tx_buf = ctx->current_tx->buf;
```

Comment:
Initialization of `ctx->current_tx` and `ctx->tx_count` is missing now, see RX counterpart.